src/io/image_reader.cpp

```diff
@@ -138,13 +135,10 @@ void ReadPngFile(std::string_view path, deepworks::Tensor& out_tensor) {
         if (out_tensor.empty()) {
             out_tensor.allocate(deepworks::Shape{height, width, channels});
         } else {
-            DeepWorks_Assert(height == out_tensor.shape()[0]);
-            DeepWorks_Assert(width == out_tensor.shape()[1]);
-            DeepWorks_Assert(channels == out_tensor.shape()[2]);
+            DeepWorks_Assert(height * width * channels == out_tensor.total());
```

Comment:
Why should we delete it? These are not equivalent checks

Reply:
Not make reshape all time, just read image to memory